src/main/java/net/coderbot/iris/mixin/MixinWorldRenderer.java

```diff
@@ -112,4 +118,15 @@
 	private void iris$endWorldBorder(Camera camera, CallbackInfo callback) {
 		Iris.getPipeline().endWorldBorder();
 	}
+
+	@Inject(method = "renderEntity", at = @At("HEAD"))
+	private void iris$getCurrentEntity(Entity entity, double cameraX, double cameraY, double cameraZ, float tickDelta, MatrixStack matrices, VertexConsumerProvider vertexConsumers, CallbackInfo ci){
+		CapturedRenderingState.INSTANCE.setCurrentEntity(entity);
+	}
+
+	@Inject(method = RENDER, at = @At(value = "INVOKE", target = "Lnet/minecraft/block/entity/BlockEntity;getPos()Lnet/minecraft/util/math/BlockPos;", ordinal = 1), locals = LocalCapture.CAPTURE_FAILHARD)
+	private void iris$getCurrentBlockEntity(MatrixStack matrices, float tickDelta, long limitTime, boolean renderBlockOutline, Camera camera, GameRenderer gameRenderer, LightmapTextureManager lightmapTextureManager, Matrix4f matrix4f, CallbackInfo ci, Profiler profiler, Vec3d vec3d, double d, double e, double f, Matrix4f matrix4f2, boolean bl, Frustum frustum2, boolean bl3, VertexConsumerProvider.Immediate immediate, Set var39, Iterator var40, BlockEntity blockEntity2){
+		CapturedRenderingState.INSTANCE.setCurrentBlockEntity(blockEntity2);
+	}
```

Comment:
These aren't properly cleared when entities *aren't*  being rendered, so the uniforms will hold unspecified values. It would probably be better for them to just hold `null`.